### PR TITLE
Small improvements to index buffer/texture loader 

### PR DESCRIPTION
**Pull request by guycalledfrank**
- Add initialData and setData to indexBuffer. Works the same way as with vertexBuffer (VB already has it).
- Allow texture loader to load from blob URLs (window.URL.createObjectURL)

### CODE REVIEW

**Comment by willeastcott** [comment]
Looks fine to me. @daredevildave?

File: src/resources/texture.js [hunk 1]
```diff
@@ -82,12 +82,32 @@ pc.extend(pc, function () {
...
             } else {
-                // Unsupported texture extension
-                // Use timeout because asset events can be hooked up after load gets called in some
-                // cases. For example, material loads a texture on 'add' event.
-                setTimeout(function () {
-                    callback(pc.string.format("Error loading Texture: format not supported: '{0}'", ext));
-                }, 0);
+                var blobStart = urlWithoutParams.indexOf("blob:");
+                if (blobStart >= 0) {
+                    urlWithoutParams = urlWithoutParams.substr(blobStart, urlWithoutParams.length - blobStart);
```
Comment (H1Gdev) [inline src/resources/texture.js]: @guycalledfrank 

Extract characters to the end of the string, so you can omit 2nd argument.
```js
urlWithoutParams = urlWithoutParams.substr(blobStart);
```

File: src/graphics/index-buffer.js [hunk 2]
```diff
@@ -15,7 +15,7 @@ pc.extend(pc, function () {
      * @param {Number} numIndices The number of indices to be stored in the index buffer.
      * @param {Number} [usage] The usage type of the vertex buffer (see pc.BUFFER_*).
```
Comment (H1Gdev) [inline src/graphics/index-buffer.js]: @guycalledfrank 
Add initialData @param.

File: src/graphics/index-buffer.js [hunk 3]
```diff
@@ -15,7 +15,7 @@ pc.extend(pc, function () {
...
      */
-    var IndexBuffer = function (graphicsDevice, format, numIndices, usage) {
+    var IndexBuffer = function (graphicsDevice, format, numIndices, usage, initialData) {
         // Initialize optional parameters
         // By default, index buffers are static (better for performance since buffer data can be cached in VRAM)
         this.usage = usage || pc.BUFFER_STATIC;
```
Comment (H1Gdev) [inline src/graphics/index-buffer.js]: @guycalledfrank 
If usage parameter omitted, set initialData to this.usage.

Comment (guycalledfrank) [reply src/graphics/index-buffer.js]: That might be a bit uselessly complicated, there are not too many arguments. And if we'd have more arguments, we could change them to be an object.

Comment (H1Gdev) [reply src/graphics/index-buffer.js]: I don't know what kind of use is supposed to be added in the future.
But many **usage** parameter of new IndexBuffer is omitted in [playcanvas-latest.js](https://code.playcanvas.com/playcanvas-latest.js).
So I thought that **usage** parameter is usually omitted.

**Pull request merged**